solution/0500-0599/0550.Game Play Analysis IV/README.md

```diff
@@ -107,4 +107,21 @@ FROM T
 WHERE rk = 1;
 ```
 
+### **pandas**
+
+```python
+import pandas as pd
+
+
+def gameplay_analysis(activity: pd.DataFrame) -> pd.DataFrame:
+    activity["first"] = activity.groupby("player_id").event_date.transform(min)
+    activity_2nd_day = activity.loc[
```

Comment:
Boolean Indexing: In the original code, the` loc `method was used to filter rows based on a boolean condition. The optimized code directly applies the condition to the DataFrame using boolean indexing: `activity[activity['first'] + pd.DateOffset(1) == activity['event_date']]`. This simplifies the code and is more idiomatic.
<img width="1318" alt="image" src="https://github.com/doocs/leetcode/assets/117748716/e3abb4fb-9240-47e9-b197-dc0558e58e73">